src/components/CommerceProfile.js

```diff
@@ -415,6 +458,85 @@ class commerceData extends Component {
     }
   };
 
+  onProvinceNameChangeOnMap = newProvinceName => {
+    this.matchProvinceByValue(newProvinceName);
+  };
+
+  matchProvinceByValue = name => {
+    const province = this.props.provincesList.find(
+      province => province.label === name
+    );
+
+    if (province) {
+      this.props.onCommerceValueChange({
+        prop: 'province',
+        value: { provinceId: province.value, name }
+      });
+    } else {
+      this.props.onCommerceValueChange({
+        prop: 'province',
+        value: { provinceId: '', name: '' }
+      });
+    }
+  };
+
+  onMapPress = () => {
+    const {
+      address,
+      provinceName,
+      city,
+      latitude,
+      longitude
+    } = this.props.locationData;
+
+    const navigateAction = NavigationActions.navigate({
+      routeName: 'changeAddressMap',
+      params: {
+        callback: this.onProvinceNameChangeOnMap,
+        markers: [
+          {
+            address,
+            provinceName,
+            city,
+            latitude,
+            longitude
+          }
+        ]
+      }
```

Comment:
todo esto se puede pasar directamente en this.props.navigation.navigate('routeName', { ...los parametros } ) y no hace falta importar NavigationActions, de ultima si no queres meter tanto entre parentesis, definis el objeto params afuera y se lo pasas